lazysnap/src/main/java/com/google/accompanist/lazysnap/SnappingFlingBehavior.kt

```diff
@@ -0,0 +1,689 @@
+import androidx.compose.foundation.gestures.ScrollScope
+import androidx.compose.foundation.lazy.LazyListItemInfo
+import androidx.compose.foundation.lazy.LazyListLayoutInfo
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import io.github.aakira.napier.DebugAntilog
+import io.github.aakira.napier.Napier
+import kotlin.math.abs
+import kotlin.math.absoluteValue
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.truncate
+
+private const val DebugLog = false
+
+@RequiresOptIn(message = "Accompanist Lazy Snap is experimental. The API may be changed in the future.")
+@Retention(AnnotationRetention.BINARY)
+annotation class ExperimentalLazySnapApi
+
+/**
+ * Default values used for [SnappingFlingBehavior] & [rememberSnappingFlingBehavior].
+ */
+@ExperimentalLazySnapApi
+object SnappingFlingBehaviorDefaults {
+    /**
+     * [AnimationSpec] used as the default value for the `snapAnimationSpec` parameter on
+     * [rememberSnappingFlingBehavior] and [SnappingFlingBehavior].
+     */
+    val snapAnimationSpec: AnimationSpec<Float> = spring(stiffness = 400f)
+
+    /**
+     * The default implementation for the `maximumFlingDistance` parameter of
+     * [rememberSnappingFlingBehavior] and [SnappingFlingBehavior], which does not limit
+     * the fling distance.
+     */
+    val maximumFlingDistance: (LazyListLayoutInfo) -> Int = { Int.MAX_VALUE }
+}
+
+/**
+ * Create and remember a snapping [FlingBehavior] to be used with [LazyListState].
+ *
+ * @param lazyListState The [LazyListState] to update.
+ * @param snapOffsetForItem Block which returns which offset the given item should 'snap' to.
+ * See [SnapOffsets] for provided values.
+ * @param maximumFlingDistance Block which returns the maximum fling distance in pixels.
+ * The returned value should be >= 0.
+ * @param decayAnimationSpec The decay animation spec to use for decayed flings.
+ * @param snapAnimationSpec The animation spec to use when snapping.
+ * @param endContentPadding The amount of content padding on the end edge of the lazy list
+ * in pixels (end/bottom depending on the scrolling direction).
+ */
+@ExperimentalLazySnapApi
+@Composable
+fun rememberSnappingFlingBehavior(
+    lazyListState: LazyListState,
+    snapOffsetForItem: (layoutInfo: LazyListLayoutInfo, itemInfo: LazyListItemInfo) -> Int = SnapOffsets.Center,
+    maximumFlingDistance: (LazyListLayoutInfo) -> Int = SnappingFlingBehaviorDefaults.maximumFlingDistance,
+    decayAnimationSpec: DecayAnimationSpec<Float> = rememberSplineBasedDecay(),
+    snapAnimationSpec: AnimationSpec<Float> = SnappingFlingBehaviorDefaults.snapAnimationSpec,
+    @Px endContentPadding: Int = 0,
+): SnappingFlingBehavior = remember(
```

Comment:
we use `snap` in the module name and in SnapOffsets class, but here we use `snapping`. maybe we can unify that, does it sound bad to just call it SnapFlingBehavior? in RV it was called SnapHelper

Reply:
Sounds good. Done.